### PR TITLE
Fix ROOT-9139

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

File: core/cont/src/TClassTable.cxx [hunk 1]
```diff
@@ -211,6 +211,7 @@ namespace ROOT {
       static std::vector<std::pair<const char *, const char *>> delayedAddClassAlternate;
       return delayedAddClassAlternate;
    }
+
```
Comment (amadio) [inline core/cont/src/TClassTable.cxx]: Spurious change?

File: core/cont/src/TClassTable.cxx [hunk 2]
```diff
@@ -255,6 +256,20 @@ TClassTable::~TClassTable()
    delete fgIdMap; fgIdMap = 0;
 }
 
+/// Return true is the table exist.
```
Comment (amadio) [inline core/cont/src/TClassTable.cxx]: `is` -> `if`

File: core/cont/src/TClassTable.cxx [hunk 3]
```diff
@@ -255,6 +256,20 @@ TClassTable::~TClassTable()
...
+/// If the table does not exist but the delayed list does, then
+/// create the table and return true.
+inline Bool_t TClassTable::CheckClassTableInit() {
+   if (!gClassTable || !fgTable) {
+      if (GetDelayedAddClass().size()) {
+         new TClassTable;
```
Comment (amadio) [inline core/cont/src/TClassTable.cxx]: Shouldn't you copy over the classes from the temporary table here?

File: core/cont/src/TClassTable.cxx [hunk 4]
```diff
@@ -518,7 +533,7 @@ TClassRec *TClassTable::FindElementImpl(const char *cname, Bool_t insert)
 
 TClassRec *TClassTable::FindElement(const char *cname, Bool_t insert)
 {
-   if (!fgTable) return 0;
+   if (!CheckClassTableInit()) return nullptr;
```
Comment (amadio) [inline core/cont/src/TClassTable.cxx]: Return statement should go on a separate line, per coding conventions.

Comment (pcanal) [reply core/cont/src/TClassTable.cxx]: Yes.  However, the surrounding code still follow the previous convention, so keeping it as is for now.

File: core/cont/src/TClassTable.cxx (hunk 3 above)
Comment (pcanal) [reply core/cont/src/TClassTable.cxx]: Yes.  This is done in the TClassTable constructor.

File: core/cont/src/TClassTable.cxx (hunk 1 above)
Comment (pcanal) [reply core/cont/src/TClassTable.cxx]: yes. removed.

File: core/cont/src/TClassTable.cxx (hunk 2 above)
Comment (pcanal) [reply core/cont/src/TClassTable.cxx]: thanks.

**Comment by pcanal** [comment]
merged by hand.

**Pull request closed without merging**